sql/schema_changer_test: Improve reliability of TestRaceWithBackfill

spilchen opened the pull request:
The TestRaceWithBackfill test sets the GC TTL threshold to 0, which causes everything to be garbage collected immediately. Occasionally, this leads to test failures with errors like:

```
pq: batch timestamp 1729337275.789296372,0 must be after replica GC threshold 1729337288.197835522,0 (r73: /Table/106/{1/160-4/40})
```
Previously, we added retries using SucceedsSoon, but this approach felt like a game of whack-a-mole, as new failure points kept emerging. Instead, I’ve set the GC TTL threshold to 1, so it's no longer immediate. The goal is to reduce the likelihood of the replica GC threshold advancing past the batch timestamp.

Fixes #132986
Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/133440)
<!-- Reviewable:end -->

rafiss reviewed (approved):
this seems like a good change to make.

i also realized that the failure was with the race detector enabled, which makes the test slower and also flakier when it comes to timing related things, so we could add a `skip.UnderRace()` call to the test as well.

spilchen committed:
sql/schema_changer_test: Improve reliability of TestRaceWithBackfill

The TestRaceWithBackfill test sets the GC TTL threshold to 0, which
causes everything to be garbage collected immediately. Occasionally,
this leads to test failures with errors like:

```
pq: batch timestamp 1729337275.789296372,0 must be after replica GC threshold 1729337288.197835522,0 (r73: /Table/106/{1/160-4/40})
```
Previously, we added retries using SucceedsSoon, but this approach felt
like a game of whack-a-mole, as new failure points kept emerging.
Instead, I’ve set the GC TTL threshold to 1, so it's no longer
immediate. The goal is to reduce the likelihood of the replica GC
threshold advancing past the batch timestamp.

Fixes #132986
Epic: None
Release note: None

spilchen commented:
> i also realized that the failure was with the race detector enabled, which makes the test slower and also flakier when it comes to timing related things, so we could add a skip.UnderRace() call to the test as well.

Good point!

spilchen commented:
TFTR!

bors r+
